File: WoofAndroid/design-system/src/main/kotlin/com/perrystreet/woof/designsystem/atomic/molecules/placeholder/MolTextBlockPlaceholder.kt
package com.perrystreet.woof.designsystem.atomic.molecules.placeholder

import androidx.compose.foundation.layout.Arrangement
import androidx.compose.foundation.layout.Column
import androidx.compose.foundation.layout.fillMaxWidth
import androidx.compose.runtime.Composable
import androidx.compose.ui.Modifier
import com.perrystreet.woof.designsystem.atomic._tokens.spacing.SpacingRoles
import com.perrystreet.woof.designsystem.atomic.atoms.placeholder.AtomPlaceholder
import com.perrystreet.woof.designsystem.atomic.atoms.placeholder.roles.PlaceholderRole
import com.perrystreet.woof.designsystem.atomic.atoms.spacer.AtomSpacer

@Composable
fun MolTextBlockPlaceholder(lines: Int = 3) {
    Column(
        modifier = Modifier.fillMaxWidth(),
        verticalArrangement = Arrangement.spacedBy(SpacingRoles.Component.Compact.dp),
    ) {
        AtomPlaceholder(role = PlaceholderRole.Title)
        AtomSpacer(spacing = SpacingRoles.Component.ExtraCompact)
        repeat(lines) {
            AtomPlaceholder(role = PlaceholderRole.Body)
        }
    }
}

File: WoofIOS/SwiftPackages/DesignSystem/Sources/DesignSystem/Atomic/Molecules/Placeholder/MolTextBlockPlaceholder.swift
import SwiftUI

public struct MolTextBlockPlaceholder: View {
    private let lines: Int

    public init(lines: Int = 3) {
        self.lines = lines
    }

    public var body: some View {
        VStack(alignment: .leading, spacing: SpacingRoles.Component.compact.rawValue) {
            AtomPlaceholder(role: .title)
            AtomSpacer(spacing: .extraCompact)
            ForEach(0..<lines, id: \.self) { _ in
                AtomPlaceholder(role: .body)
            }
        }
        .frame(maxWidth: .infinity, alignment: .leading)
    }
}
